src/Http/Http.Abstractions/src/Validation/ValidatablePropertyInfo.cs

```diff
@@ -13,12 +16,13 @@ namespace Microsoft.AspNetCore.Http.Validation;
 public abstract class ValidatablePropertyInfo : IValidatableInfo
 {
     private RequiredAttribute? _requiredAttribute;
+    private readonly bool _hasDisplayAttribute;
 
     /// <summary>
     /// Creates a new instance of <see cref="ValidatablePropertyInfo"/>.
     /// </summary>
     protected ValidatablePropertyInfo(
-        [param: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicProperties)]
+        [param: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicProperties | DynamicallyAccessedMemberTypes.PublicConstructors)]
```

Comment:
Why is this necessary?

Reply:
We do constructor lookup in `HasDisplayAttribute` to make sure that we are respecting `[Display]` attributes on constructor parameters for records.